Sources/SwiftDiagnostics/DiagnosticDecorators/ANSIDiagnosticDecorator.swift

```diff
@@ -0,0 +1,248 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+extension DiagnosticDecorator where Self == ANSIDiagnosticDecorator {
+  @_spi(Testing) public static func ANSI(colorize: Bool) -> Self {
+    Self(colorize: colorize)
+  }
+}
+
+/// An implementation of ``DiagnosticDecorator`` protocol designed to enhance diagnostic messages by adding severity-based
+/// prefixes and optional ANSI colorization.
+///
+/// By leveraging ANSI codes—a set of control characters used to manipulate the formatting
+/// and appearance of text in a terminal—the decorator imbues textual information with visual cues.
+/// These visual cues vary depending on the severity of the message, such as whether it's an error, a warning,
+/// or another type of diagnostic output.
+///
+/// In essence, this decorator serves two primary functions:
+/// 1. It appends a categorization prefix (e.g., "error: ", "warning: ", "note: ", "remark: ") to the diagnostic message
+/// to indicate its level of urgency or importance.
+/// 2. It has the capability to colorize this prefix and other segments of the message for enhanced visibility
+/// and quick interpretation. This colorization feature can be toggled on or off, offering flexibility in its usage.
+@_spi(Testing) public struct ANSIDiagnosticDecorator: DiagnosticDecorator {
+  /// A flag indicating whether or not to apply colorization.
+  private let colorize: Bool
```

Comment:
Instead of having a `colorize` flag here, shouldn’t we just have two decorators? One that does add color and one that does not?

Reply:
Great idea! Thanks! Implemented!

Reply:
🫶